Redesign Paginator

mehrdadkhoddami opened the pull request:
- Reformat generateButton to get label
- define Offset for range
- Refactor generateRange to have better algorithm and consider Offsets
- Refactor generateKeyboard to work with this new solution
- Add optional setMaxPageBasedOnLabels to generate max_page according to labels

mehrdadkhoddami committed:
Redesign Paginator
- Reformat generateButton to get label
- define Offset for range
- Refactor generateRange to have better algorithm and consider Offsets
- Refactor generateKeyboard to work with this new solution
- Add optional setMaxPageBasedOnLabels to generate max_page according to labels

noplanman commented:
Hi, thanks for your contribution.

Please make sure you adhere to the PSR-2 guidelines, in your case 4 spaces instead of tabs.

Then I can take a look and see what your actual addition is 👍

mehrdadkhoddami committed:
fix PSR-2 format issue

mehrdadkhoddami commented:
Hi,
Sure

Notes with thanks.
I reformat file with PSR-2 . But I didn't change tests.

noplanman commented:
I can have a look at the tests and see what's up.

Before that though, could you please explain in more depth what this PR does exactly?
Best show some examples of what is new / changed.

Thanks!

mehrdadkhoddami commented:
Sure.

* First of all, I add a range for selected page. e.g if we have 9 pages, selected page is 6 & range for it was set 1, in the pagination, user can see 5 & 7 (default is 1 as range).
[img](https://drive.google.com/open?id=0B9W5NOT_Qwc5TDEwNGhlS1dDalE)
In this case, I rewrite generateRange() method.

* Also, User can pass only labels which he/she wants. (Same as before)
but now user can call setMaxPageBasedOnLabels() method to set $max_buttons according to passed labels & and range of selected page.

* I changed generateButton() to get label of button.

* I added two conditions to setSelectedPage() :
1. If user requests negative or zero page number, method sets it to page #1 automatically.
2. If user requests page which is greater than total number of pages, method sets it to total number of pages automatically.

* Finally I refactor generateKeyboard() based on these changes.

Thank you.

mehrdadkhoddami committed:
Change name of setMaxPageBasedOnLabels() to setMaxButtonsBasedOnLabels()
